### PR TITLE
Fix and document RDS bastion

### PR DESCRIPTION
**Pull request by ethanmills**

### CODE REVIEW

File: doc/connecting-to-hosted-databases.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Connecting to hosted databases
```
Comment (ethanmills) [inline doc/connecting-to-hosted-databases.md:1]: Recommend using the "rich diff" view of this in the top right if you've not seen it before

File: terraform/modules/data-share-service/ec2.tf [hunk 2]
```diff
@@ -1,44 +1,31 @@
...
+    values = ["amzn2-ami-kernel-*-x86_64-gp2"]
   }
-
   filter {
     name   = "virtualization-type"
     values = ["hvm"]
   }
-
-  owners = ["099720109477"] # Canonical
-}
```
Comment (ethanmills) [inline terraform/modules/data-share-service/ec2.tf:15]: No specific reason for this, was just marginally easier to get the latest version, and seemed like a sensible default

File: terraform/modules/data-share-service/ec2.tf [hunk 3]
```diff
@@ -1,44 +1,31 @@
...
-    http_endpoint = "disabled"
+    http_endpoint = "enabled"
+    http_tokens   = "required"
   }
   root_block_device {
     encrypted = true
   }
   tags = {
-    Name = "${var.environment}-rds-bastion"
+    Name = "${var.environment}-rds-bastion-host"
```
Comment (ethanmills) [inline terraform/modules/data-share-service/ec2.tf:28]: We both added this in parallel, I have no preference between the two

File: terraform/modules/data-share-service/ec2.tf [hunk 4]
```diff
@@ -79,3 +66,7 @@ module "tls_private_key" {
   name   = "rds-bastion-key-${var.environment}"
   source = "github.com/hashicorp-modules/tls-private-key?ref=5918adb7efe39d7b36f0185569684b1b73f18126"
 }
+
+resource "aws_eip" "rds_bastion_ip" {
+  instance = aws_instance.rds_bastion_host.id
```
Comment (ethanmills) [inline terraform/modules/data-share-service/ec2.tf:71]: Be nice to have a persistent IP for when e.g. the instance AMI updates

**Review by oskarwilliams378 — APPROVED** [review]

**Review by olevett — APPROVED** [review]

**Pull request merged**